cmd/main.go

```diff
@@ -170,6 +170,18 @@ func main() {
 	}
 }
 
+func getQueueClient(cfg config.Configuration) queue.QueueClient {
+	var queueClient queue.QueueClient
+	if cfg.Queue.Type == config.RedisQueueProvider {
+		queueClient = convoyRedis.NewQueueClient()
+	}
+	if cfg.Queue.Type == config.InMemoryQueueProvider {
+		queueClient = convoyMemqueue.NewQueueClient()
+	}
+
+	return queueClient
```

Comment:
Should we make allowance for an error here, or does the caller have to deduce that on their own?

Reply:
I'm currently refactoring.